src/ChannelServer/World/Map.cs

```diff
@@ -251,6 +251,112 @@ public List<ICombatEntity> GetAttackableEntitiesInRange(ICombatEntity attacker,
 			return result;
 		}
 
+		/// <summary>
+		/// Returns attackable monsters sorted by their distance in the given 
+		/// radius around position.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public List<ICombatEntity> GetAttackableEntitiesInRangeSortedByDistance(Character attacker, Position position, float radius)
+		{
+			var result = new List<ICombatEntity>();
+
+			lock (_monsters)
+			{
+				var entities = _monsters.Values.Where(a => a.Position.InRange2D(position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(position));
+				result.AddRange(entities);
+			}
+
+			lock (_characters)
+			{
+				var entities = _characters.Values.Where(a => a.Position.InRange2D(position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(position));
+				result.AddRange(entities);
+			}
+
+			return result;
+		}
+
+		/// <summary>
+		/// Returns attackable monsters sorted by their distance in the given 
+		/// radius around position.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public List<ICombatEntity> GetAttackableEntitiesInRangeAroundEntity(Character attacker, IEntity entity, float radius)
+		{
+			var result = new List<ICombatEntity>();
+
+			lock (_monsters)
+			{
+				var entities = _monsters.Values.Where(a => entity.Handle != a.Handle && a.Position.InRange2D(entity.Position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(entity.Position));
+				result.AddRange(entities);
+			}
+
+			lock (_characters)
+			{
+				var entities = _characters.Values.Where(a => entity.Handle != a.Handle && a.Position.InRange2D(entity.Position, radius) && attacker.CanAttack(a)).OrderBy(a => a.Position.Get2DDistance(entity.Position));
+				result.AddRange(entities);
+			}
+
+			return result;
+		}
+
+
+		/// <summary>
+		/// Returns attackable monsters sorted by their distance in the given 
+		/// radius around position.
+		/// </summary>
+		/// <param name="attacker"></param>
+		/// <param name="position"></param>
+		/// <param name="radius"></param>
+		/// <returns></returns>
+		public ICombatEntity GetNearestAttackableEntity(Character attacker, Position position, float radius)
```

Comment:
And here you wouldn't get the actual nearest entity at all if you have both monsters and characters as options.

Reply:
Similar to the comment above, these are skill-related methods I made, to emulate the behavior I saw when skills are cast on the server, but like you said I add the monsters first and sort them and then characters and sort them. Making the code prioritize monsters first. So it was a rough implementation since the state of Melia is PVE right now.

Reply:
Removed skill-related changes, they aren't supposed to be a part of this PR.

Reply:
>it was a rough implementation since the state of Melia is PVE right now

Yea, that's fine. Though I would suggest to generally keep the future in mind, and to implement things in a way that (hopefully) won't require much changes in the future. That way you save yourself the refactoring, or the potential confusion, when you or someone else calls this method, expecting certain results, and getting something else. In this particular case, it wouldn't even have required much more code after all.